src/Service/GdsService.php

```diff
@@ -0,0 +1,372 @@
+        GatewayResourceService $resourceService,
+        FileSystemHandleService $fshService,
+        BrkService $brkService,
+        CacheService $cacheService,
+        EntityManagerInterface $entityManager,
+        EventDispatcherInterface $eventDispatcher
+    ) {
+        $this->resourceService = $resourceService;
+        $this->fshService      = $fshService;
+        $this->callService     = $callService;
+        $this->brkService      = $brkService;
+        $this->cacheService    = $cacheService;
+        $this->entityManager   = $entityManager;
+        $this->eventDispatcher = $eventDispatcher;
+
+    }//end __construct()
+
+
+    /**
+     * Creates the content for a soap message to the GDS download service.
+     *
+     * @param DateTime $lastSynced    The date and time the data has last been synced.
+     * @param array    $configuration Whether or not the call is a test call.
+     *
+     * @return array The resulting request message.
+     */
+    private function createRequestMessage(DateTime $lastSynced, array $configuration): array
+    {
+        $now            = new DateTime();
+        $notYetReported = 'true';
+
+        if ($configuration['test'] === true) {
+            $notYetReported = 'false';
+        }
+
+        return [
+            "@xmlns:SOAP-ENV" => "http://schemas.xmlsoap.org/soap/envelope/",
+            "@xmlns:v20"      => "http://www.kadaster.nl/schemas/gds2/service/afgifte-bestandenlijstopvragen/v20170401",
+            "@xmlns:v201"     => "http://www.kadaster.nl/schemas/gds2/afgifte-bestandenlijstselectie/v20170401",
+            "@xmlns:v202"     => "http://www.kadaster.nl/schemas/gds2/afgifte-proces/v20170401",
+            "SOAP-ENV:Body"   => [
+                "v20:BestandenlijstOpvragenRequest" => [
+                    "v20:verzoek" => [
+                        "v201:AfgifteSelectieCriteria" => [
+                            'v201:BestandKenmerken'     => ["v201:contractnummer" => "{$configuration['contractNumber']}"],
+                            "v202:Periode"              => [
+                                "v202:DatumTijdVanaf"  => $lastSynced->format('Y-m-d\TH:i:s.v\Z'),
+                                "v202:DatumTijdTotmet" => $now->format('Y-m-d\TH:i:s.v\Z'),
+                            ],
+                            "v202:NogNietGerapporteerd" => $notYetReported,
+                            "v202:Sortering"            => [
+                                "v202:Kolom"    => "DATUM_AANMELDING",
+                                "v202:Volgorde" => "DESC",
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+    }//end createRequestMessage()
+
+
+    /**
+     * Determines if an array is associative.
+     *
+     * @param array $array The array to check.
+     *
+     * @return bool Whether the array is associative.
+     */
+    private function isAssociative(array $array): bool
```

Comment:
I have seen this function before, a few times. Can we re-use an already existing one and remove this function instead?

Reply:
I will remove this one, as it also is present in the BRKservice, but indeed we should add it somewhere in core or something like that.